### PR TITLE
[Ready] Emp Proof Cells! (Best Grammar no conflict merges)

### PR DESCRIPTION
**Pull request by Trilbyspaceclone**
[Changelogs]
Adds in a new tech node that cost 5000 Sci points to get EMP proof cells
Adds in EMP proof cells form High power to Blue space cells, craft-able via prolathens in Sci/Engi departments
Down side to these cells being EMP proof is that they cost PLASMA! Also they hold up to HALF the normal power stored in the non-emp proof cells!
[why]
EMP cells are all ready a thing, EMP lets face it is OP as hell. What better way to help curve the power of it then to counter it with options and content and like things!

### CODE REVIEW

**Comment by Trilbyspaceclone** [comment]
**PROOF THAT IT ALL WORKS AND ITS ALL TESTED AND NOT JUST WEB _EDITS**
https://cdn.discordapp.com/attachments/455477249299185704/498651891090194432/unknown.png
https://cdn.discordapp.com/attachments/455477249299185704/498651780243128330/unknown.png
https://cdn.discordapp.com/attachments/435484634751434754/498656078054031405/unknown.png
It works... It all works damn it... Its tested...

**Review by deathride58 — CHANGES_REQUESTED** [review]
Needs to use the emp protection component rather than a snowflake var

**Comment by deathride58** [comment]
EMPs are supposed to be a hard counter to all electronic devices, and are already pretty underused as-is due to how uncontrollable their AoE effects tend to be. There's also cases where EMPs are the ***only*** hard counter to something, such as with mechs, which can only really be reliably killed by EMPs, as EMPs drain the cell the mech requires to operate.

**Comment by Trilbyspaceclone** [comment]
" emp protection component " 
Ok lets look at how these work 
EMP protection self = Makes it when cells are at 0 charge and EMP there not rigged to blow up when charged and used
EMP protection components = Protects the chems/items inside the object but not power charge!
EMP still works on mechs to drain charge, still works on EMPing stun prod!
Why still have them? For APCs, and other items that EMP like circuits. EMP a APCs and it only messes with wires not the cell so it can more easily recover. EMP in cyborgs are a unknown but I would hope they work in them so its not "Welp now im powerless after 2 shots forma flashlight hardstuned and cant talk..."

**Review by deathride58 — CHANGES_REQUESTED** [review]
The AddComponent()s need to be moved from the /empty subtypes to the base subtypes. To reduce copypasta further, you can make an emp-proof type that all the other emp-proof cells inherit from

File: code/modules/research/techweb/all_nodes.dm [hunk 1]
```diff
@@ -332,6 +332,14 @@
 	research_costs = list(TECHWEB_POINT_TYPE_GENERIC = 3000)
 	export_price = 5000
 
+/datum/techweb_node/emp_proof
+	id = "emp_proof"
+	display_name = "EMP Proof Cells"
+	description = "Highly advanced cells that are not as affected by EMP, tho can't store as much power as their unprotected counter parts."
```
Comment (silicons) [inline code/modules/research/techweb/all_nodes.dm]: gramar

**Comment by nik707** [comment]
@deathride58 This pass your requested changes yet, or does it need more fixes to be ready for merge

**Comment by Trilbyspaceclone** [comment]
Its all ready

**Comment by improvedname** [comment]
Emp proof cells is a thing because KA.

**Comment by silicons** [comment]
Unfortunately I do not believe this to be in our best interests for science balance. The whole idea of mass producible "Counters the hard counter to this" is against what I want for science, even if guns don't apply here.

**Pull request closed without merging**